src/Http/RequestFactory.php

```diff
@@ -194,30 +194,54 @@ public function createHttpRequest()
 		$usingTrustedProxy = $remoteAddr && array_filter($this->proxies, function ($proxy) use ($remoteAddr) {
 			return Helpers::ipMatch($remoteAddr, $proxy);
 		});
-
 		if ($usingTrustedProxy) {
-			if (!empty($_SERVER['HTTP_X_FORWARDED_PROTO'])) {
-				$url->setScheme(strcasecmp($_SERVER['HTTP_X_FORWARDED_PROTO'], 'https') === 0 ? 'https' : 'http');
-			}
+			if(!empty($_SERVER['HTTP_FORWARDED'])) {
+				$forwardParams = preg_split('/[,]|[;]/', $_SERVER['HTTP_FORWARDED']);
+				foreach ($forwardParams as $forwardParam) {
+					$param = explode("=", $forwardParam);
+					$proxyParams[trim($param[0])][] = trim($param[1]);   //e.g. array['for'][0] = 192.168.0.1
```

Comment:
Parameter names must be treated in case-insensitive manner.
solved by commit 44660bd